### PR TITLE
Add new guide/summary for multihost testing

### PR DESCRIPTION
**Pull request by happz**
Includes small fixes in other parts of documentation related to changes in multihost support.

### CODE REVIEW

**Review by psss — CHANGES_REQUESTED** [review]
Thanks much for writing this! This will be extremely helpful for new users. I've noticed only a couple of minor typos. As this is an advanced topic, I'd suggest to move it to the bottom of the guide so that the `Under The Hood` which explains more basic stuff is read first.

File: docs/guide.rst [hunk 1]
```diff
@@ -142,6 +142,126 @@ to refresh config to the latest version.
 __ https://pre-commit.com/#install
 
 
+Multihost testing
```
[the remark this answers is not in the document]
Comment (happz) [reply docs/guide.rst]: Fixed and moved to the end of the guide in 52650a1df74c66aa27bdec0104f4d89606e65c3a

File: docs/guide.rst [hunk 2]
```diff
@@ -142,6 +142,126 @@ to refresh config to the latest version.
...
+.. versionadded:: 1.24
+
+Support for basic server/client scenarios is now available.
+
+``prepare``, ``execute``, and ``finish`` steps are able to run a
+given task (test, preparation script, ansible playbook, etc.) on
+several guests at once. Tasks are assigned to provisioned guests by
+matching the ``where`` key from
+:ref:`preparation</spec/plans/discover/where>`,
+:ref:`preparation</spec/plans/prepare/where>` and ``finish`` phases
```
[the remark this answers is not in the document]
Comment (happz) [reply docs/guide.rst]: Fixed in 52650a1df74c66aa27bdec0104f4d89606e65c3a

**Review by psss — APPROVED** [review]
Thanks for addressing all comments!

**Pull request merged**